### PR TITLE
add support for `FOR ORDINALITY` and `NESTED` in JSON_TABLE

### PR DESCRIPTION
**Pull request by lovasoa**
fixes https://github.com/apache/datafusion-sqlparser-rs/issues/1492

see the specs here: https://dev.mysql.com/doc/refman/8.0/en/json-table-functions.html#function_json-table

@alamb : this is a fix for the following bug in SQLPage: https://github.com/sqlpage/SQLPage/issues/658

### CODE REVIEW

File: src/ast/query.rs [hunk 1]
```diff
@@ -2281,14 +2281,65 @@ impl fmt::Display for ForJson {
 /// FROM JSON_TABLE(
```
Comment (iffyio) [inline src/ast/query.rs:2281]: Could we include the link to the mysql doc here?

**Review by iffyio — APPROVED** [review]
LGTM! cc @alamb

**Comment by alamb** [comment]
Thanks @lovasoa and @iffyio

**Comment by lovasoa** [comment]
Thank you for merging, @alamb :blush: 

Could you maybe also take a look at https://github.com/apache/datafusion-sqlparser-rs/pull/1490 ?

**Pull request merged**